sound/soc/sof/sof-audio.c

```diff
@@ -82,6 +82,15 @@ static int sof_widget_kcontrol_setup(struct snd_sof_dev *sdev, struct snd_sof_wi
 	return 0;
 }
 
+static void sof_reset_route_setup_status(struct snd_sof_dev *sdev, struct snd_sof_widget *widget)
+{
+	struct snd_sof_route *sroute;
+
+	list_for_each_entry(sroute, &sdev->route_list, list)
+		if (sroute->src_widget == widget || sroute->sink_widget == widget)
+			sroute->setup = false;
```

Comment:
i am pretty sure I've seen this before, is this a bad squash?

Reply:
No, this is legit. What you saw before was applicable only for static pipelines. This is where the route status is updated for dynamic pipelines.